File: Definitions/Game_Definitions/classes/Wardrobe.lua
---@meta
---@diagnostic disable

---@class Wardrobe : InteractiveDevice
Wardrobe = {}

---@return Wardrobe
function Wardrobe.new() return end

---@param props table
---@return Wardrobe
function Wardrobe.new(props) return end

---@return Bool
function Wardrobe:OnGameAttached() return end

---@param evt gameinteractionsInteractionActivationEvent
---@return Bool
function Wardrobe:OnInteractionActivated(evt) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function Wardrobe:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function Wardrobe:OnTakeControl(ri) return end

---@return EGameplayRole
function Wardrobe:DeterminGameplayRole() return end

---@param data SDeviceMappinData
---@return EMappinVisualState
function Wardrobe:DeterminGameplayRoleMappinVisuaState(data) return end

---@return WardrobeController
function Wardrobe:GetController() return end

---@return WardrobeControllerPS
function Wardrobe:GetDevicePS() return end

---@return Bool
function Wardrobe:IsWardrobe() return end

function Wardrobe:ResolveGameplayState() return end

function Wardrobe:RestoreDeviceState() return end

File: DecompiledGameScripts/cyberpunk/devices/wardrobe/wardrobe.swift

public class Wardrobe extends InteractiveDevice {

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as WardrobeController;
  }

  protected cb func OnGameAttached() -> Bool {
    super.OnGameAttached();
  }

  protected func RestoreDeviceState() -> Void {
    super.RestoreDeviceState();
  }

  protected func ResolveGameplayState() -> Void {
    super.ResolveGameplayState();
  }

  protected cb func OnInteractionActivated(evt: ref<InteractionActivationEvent>) -> Bool {
    let actorUpdateData: ref<HUDActorUpdateData>;
    super.OnInteractionActivated(evt);
    if Equals(evt.eventType, gameinteractionsEInteractionEventType.EIET_activate) {
      if Equals(evt.layerData.tag, n"LogicArea") {
        actorUpdateData = new HUDActorUpdateData();
        actorUpdateData.updateIsInIconForcedVisibilityRange = true;
        actorUpdateData.isInIconForcedVisibilityRangeValue = true;
        this.RequestHUDRefresh(actorUpdateData);
      };
    } else {
      if Equals(evt.layerData.tag, n"LogicArea") {
        actorUpdateData = new HUDActorUpdateData();
        actorUpdateData.updateIsInIconForcedVisibilityRange = true;
        actorUpdateData.isInIconForcedVisibilityRangeValue = false;
        this.RequestHUDRefresh(actorUpdateData);
      };
    };
  }

  private const func GetController() -> ref<WardrobeController> {
    return this.m_controller as WardrobeController;
  }

  public const func GetDevicePS() -> ref<WardrobeControllerPS> {
    return this.GetController().GetPS();
  }

  public const func IsWardrobe() -> Bool {
    return true;
  }

  public const func DeterminGameplayRoleMappinVisuaState(const data: script_ref<SDeviceMappinData>) -> EMappinVisualState {
    return EMappinVisualState.Default;
  }

  public const func DeterminGameplayRole() -> EGameplayRole {
    if this.GetDevicePS().HasInteraction() {
      return EGameplayRole.Wardrobe;
    };
    return EGameplayRole.None;
  }
}

public class WardrobeController extends ScriptableDeviceComponent {

  public const func GetPS() -> ref<WardrobeControllerPS> {
    return this.GetBasePS() as WardrobeControllerPS;
  }
}
